cpp/basix/finite-element.cpp

```diff
@@ -533,6 +615,140 @@ xt::xtensor<double, 3> FiniteElement::map_pull_back(
   return U;
 }
 //-----------------------------------------------------------------------------
+void FiniteElement::permute_dofs(tcb::span<int>& dofs,
+                                 std::uint32_t cell_info) const
+{
+  if (!_dof_transformations_are_permutations)
+    throw std::runtime_error(
+        "The DOF transformations for this element are not permutations");
+  if (_dof_transformations_are_identity)
+    return;
+
+  const int tdim = cell::topological_dimension(_cell_type);
```

Comment:
Should the `FiniteElement` object store the `tdim` to avoid calling this function a lot?